src/Sarif.ValidationTests/SarifValidatorTests.cs

```diff
@@ -28,6 +30,71 @@ public SarifValidatorTests()
             _schema = SchemaReader.ReadSchema(schemaText, JsonSchemaFile);
         }
 
+        //[Fact]
+        internal void ValidateAllTheThings()
+        {
+            // First, we start with builders that only populate required properties that are backed by primitives.
+            IDictionary<Type, DefaultObjectPopulatingVisitor.PrimitiveValueBuilder> propertyValueBuilders =
+                DefaultObjectPopulatingVisitor.GetBuildersForAllPrimitives();
+
+            ValidateDefaultDocument(propertyValueBuilders);
+
+            // TODO will build out population mechanism to fix these remaining validation errors in the next change.
+            //
+            // error JSON1015: at "runs[0].tool.fileVersion": The string "{Optional string value}" does not match the regular expression "[0-9]+(\.[0-9]+){3}".
+            // error JSON1015: at "runs[0].conversion.tool.fileVersion": The string "{Optional string value}" does not match the regular expression "[0-9]+(\.[0-9]+){3}".
+            // error JSON1015: at "runs[0].files.PlaceholderDictionaryKey.mimeType": The string "{Optional string value}" does not match the regular expression "[^/]+/.+".
+        }
+
+        [Fact]
+        public void ValidatesUriConversion()
+        {
+            // First, we start with builders that only populate required properties that are backed by primitives.
+            IDictionary<Type, DefaultObjectPopulatingVisitor.PrimitiveValueBuilder> propertyValueBuilders =
+                DefaultObjectPopulatingVisitor.GetBuildersForRequiredPrimitives();
```

Comment:
>GetBuildersForRequiredPrimitives [](start = 47, length = 32)

Bug? You want to validate _all_ URI-valued properties, but not all of them are required. (_e.g._, `tool.resourceLocation`, `rule.helpUri`, `result.hostedViewerUri`, ...)